TruePath/PathExtensions.cs

```diff
@@ -0,0 +1,69 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace TruePath;
+/// <summary>
+/// Extension methods for <see cref="IPath"/> and <see cref="IPath{TPath}"/>.
+/// </summary>
+public static class PathExtensions
+{
+    /// <summary>
+    /// Gets the extension of the file name of the path.
+    /// </summary>
+    /// <typeparam name="TPath"></typeparam>
+    /// <param name="path"></param>
+    /// <returns></returns>
+    public static string GetExtension<TPath>(this TPath path) where TPath : IPath
```

Comment:
I don't think a generic `TPath` is needed here. You can use `IPath` directly:
```suggestion
    public static string GetExtension(this IPath path)
```

This applies to the rest of the methods as well